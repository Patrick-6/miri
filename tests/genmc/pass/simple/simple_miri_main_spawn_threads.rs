//@compile-flags: -Zmiri-genmc

#![no_main]

const N: usize = 1;

#[unsafe(no_mangle)]
fn miri_start(_argc: isize, _argv: *const *const u8) -> isize {
    let handles: Vec<_> = (0..N).map(|_| std::thread::spawn(thread_func)).collect();
    handles.into_iter().for_each(|handle| handle.join().unwrap());

    0
}

fn thread_func() {}
