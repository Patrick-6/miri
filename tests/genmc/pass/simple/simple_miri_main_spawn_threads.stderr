warning: GenMC currently does not model the atomic failure ordering for `compare_exchange`. Failure ordering 'Relaxed' is treated like 'Acquire', which means that Miri might miss bugs related to this memory access (possible unsoundness)!

warning: GenMC mode currently does not model spurious failures of `compare_exchange_weak`. This may lead to missed bugs (possible unsoundness)!

warning: GenMC currently does not model the atomic failure ordering for `compare_exchange`. Failure ordering 'Acquire' is treated like 'Relaxed', which means that Miri might incorrectly detect errors related to this memory access (possible false positives).


(GenMC) Verification complete. No errors were detected.
Number of complete executions explored: 1
